cli/src/main/java/io/aklivity/zillabase/cli/internal/migrations/repository/MigrationMetadataRepository.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2024 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zillabase.cli.internal.migrations.repository;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.postgresql.jdbc.PreferQueryMode;
+
+import io.aklivity.zillabase.cli.internal.migrations.model.ZillabaseMigrationFile;
+import io.aklivity.zillabase.cli.internal.migrations.model.ZillabaseMigrationMetadata;
+
+public final class MigrationMetadataRepository
+{
+    private final String url;
+    private final Properties props;
+
+    public MigrationMetadataRepository(
+        String dbName)
+    {
+        this.url = "jdbc:postgresql://localhost:4567/%s".formatted(dbName);
```

Comment:
Can we parameterize this based on configuration so that `ZillabaseAdminConfig` and port number here are always guaranteed to be in sync?